underpants.js

```diff
@@ -158,20 +256,49 @@ var _ = {};
 *   use _.each in your implementation
 */
 
+// accepts an array and a function
+// call func for each elem in array
+// w/ arguments of element, its index, <array>
+// return new array for which calling function returned true
+
+_.filter = (arr, func) => {
+    const newArr = [];
+    _.each(arr, function (elem, index, arr2) {
+        if (func(elem, index, arr2)) {
+            newArr.push(elem);
+        }
+    });
+    return newArr;
+};
 
 /** _.reject
 * Arguments:
 *   1) An array
 *   2) A function
 * Objectives:
 *   1) call <function> for each element in <array> passing the arguments:
-*      the element, it's index, <array>
+*      the element, its index, <array>
 *   2) return a new array of elements for which calling <function> returned false
 *   3) This is the logical inverse if _.filter()
 * Examples:
 *   _.reject([1,2,3,4,5], function(e){return e%2 === 0}) -> [1,3,5]
 */
 
+// accepts an array and a function
+// call func for each elem
+// w/ arguments of element, index, array
+// return new array of elems for which calling function returned false
+// opposite of _.filter()
+
+_.reject = (arr, func) => {
+    const newArr = [];
+    _.each(arr, function (elem, index, arr2) {
+        if (!(func(elem, index, arr2))) {
+            newArr.push(elem);
+        }
+    });
+    return newArr;
```

Comment:
Because this code is almost an exact copy of `_.filter`, consider how you could use `_.filter` here